falcon_test.go

```diff
@@ -109,28 +108,232 @@ func TestFalcon(t *testing.T) {
+			t.Fatalf("verify_ct failed err msg %s on pk: %v , sk: %v, msg: %v", err, pub, priv, msg)
 		}
 	}
 }
+
+func TestFalconCompressedSignatureSizes(t *testing.T) {
+	seed := make([]byte, 64)
+	rand.Read(seed)
+
+	pub, priv, err := GenerateKey(seed)
+	if err != nil {
+		t.Fatalf("failed to generate keys. err message: %s", err)
+	}
+
+	msg := make([]byte, 500)
+	rand.Read(msg)
+
+	sig, err := priv.SignCompressed(msg)
+	if err != nil {
+		t.Fatalf("failed to sign message. err message: %s", err)
+	}
+
+	var sig2 [SignatureMaxSize + 1]byte
+	copy(sig2[:], sig)
+	err = pub.Verify(sig2[:], msg)
+	if err == nil {
+		t.Fatalf("verification succeeded. should have failed.")
+	}
+
+	var sig3 [1]byte
+	copy(sig3[:], sig)
+	err = pub.Verify(sig3[:], msg)
+	if err == nil {
+		t.Fatalf("verification succeeded. should have failed.")
+	}
+
+	var sig4 [2]byte
+	copy(sig4[:], sig)
+	err = pub.Verify(sig4[:], msg)
+	if err == nil {
+		t.Fatalf("verification succeeded. should have failed.")
+	}
+
+}
+
+func TestFalconSignNilMessage(t *testing.T) {
+	seed := make([]byte, 64)
+	rand.Read(seed)
+
+	pub, priv, err := GenerateKey(seed)
+	if err != nil {
+		t.Fatalf("failed to generate keys. err message: %s", err)
+	}
+
+	sig, err := priv.SignCompressed(nil)
+	if err != nil {
+		t.Fatalf("failed to sign message. err message: %s", err)
+	}
+
+	err = pub.Verify(sig, nil)
+	if err != nil {
+		t.Fatalf("failed to verify message. err message: %s", err)
+	}
+
+	err = pub.Verify(sig, []byte{})
+	if err != nil {
+		t.Fatalf("failed to verify message. err message: %s", err)
+	}
+
+	ctSignature, err := sig.ConvertToCT()
+	if err != nil {
+		t.Fatalf("failed to verify message. err message: %s", err)
+	}
+
+	err = pub.VerifyCTSignature(ctSignature, nil)
+	if err != nil {
+		t.Fatalf("failed to verify message. err message: %s", err)
+	}
+
+	err = pub.VerifyCTSignature(ctSignature, []byte{})
+	if err != nil {
+		t.Fatalf("failed to verify message. err message: %s", err)
+	}
+}
+
+func TestFalconGenerateKeysDifferentSeed(t *testing.T) {
+	seed := make([]byte, 64)
+	rand.Read(seed)
+
+	pub, sk, err := GenerateKey(seed)
+	if err != nil {
+		t.Fatalf("failed to generate keys. err message: %s", err)
+	}
+
+	rand.Read(seed)
+	pub2, sk2, err := GenerateKey(seed)
+	if err != nil {
+		t.Fatalf("failed to generate keys. err message: %s", err)
+	}
+
+	if pub == pub2 {
+		t.Fatalf("public keys are the same")
+	}
+
+	if sk == sk2 {
+		t.Fatalf("private keys are the same")
+	}
+}
+
+func TestFalconNilSignature(t *testing.T) {
+	seed := make([]byte, 64)
+	rand.Read(seed)
+
+	pub, _, err := GenerateKey(seed)
+	if err != nil {
+		t.Fatalf("failed to generate keys. err message: %s", err)
+	}
+
+	msg := make([]byte, 500)
+	rand.Read(msg)
+
+	err = pub.Verify(nil, msg)
+	if err == nil {
+		t.Fatalf("verification succeeded. should have failed.")
+	}
+
+	err = pub.Verify([]byte{}, msg)
+	if err == nil {
+		t.Fatalf("verification succeeded. should have failed.")
+	}
+}
+
+func TestFalconNilSeed(t *testing.T) {
```

Comment:
two issues:
1. `%s` -> `%v`
2. when writing the error message, make sure to include what were the input parameters to GenerateKey that led to this error.